src/scripts/main.js

```diff
@@ -1 +1,69 @@
 'use strict';
+
+document.oncontextmenu = () => false;
+
+const buttonState = { left: false, right: false };
+let resetTimeout;
+
+const promise1 = new Promise((resolve, reject) => {
+  document.addEventListener('mousedown', (e) => {
+    if (e.buttons === 1) {
+      return resolve('First');
+    }
+  });
+
+  setTimeout(() => reject(new Error('First promise was rejected')), 3000);
+});
+
+const promise2 = new Promise((resolve) => {
+  document.addEventListener('mousedown', (e) => {
+    if (e.buttons === 1 || e.buttons === 2) {
+      return resolve('Second');
+    }
+  });
+});
+
+const promise3 = new Promise((resolve) => {
+  document.addEventListener('mousedown', (e) => {
+    if (e.button === 0) {
+      buttonState.left = true;
+    } else if (e.button === 2) {
+      buttonState.right = true;
+    }
+
+    if (buttonState.left && buttonState.right) {
+      resolve('Third');
+      buttonState.left = false;
+      buttonState.right = false;
+    }
+
+    clearTimeout(resetTimeout);
+
+    resetTimeout = setTimeout(() => {
+      buttonState.left = false;
+      buttonState.right = false;
+    }, 500);
```

Comment:
The timeout in promise3 resets the button states after 500ms. This might lead to unexpected behavior if the user presses both buttons with a delay greater than 500ms. Consider adjusting the timeout duration based on the expected user interaction.